src/primitives.rs

```diff
@@ -281,9 +288,75 @@ where
     }
 }
 
-impl<T> Versionize for Vec<T>
+macro_rules! impl_versionize_vec_like_type {
+    ($VecType:ident <$GenericParam:ident>) => {
+        impl<$GenericParam> Versionize for $VecType<$GenericParam>
+        where
+            $GenericParam: Versionize,
+        {
+            #[inline]
+            fn serialize<W: std::io::Write>(
+                &self,
+                mut writer: &mut W,
+                version_map: &VersionMap,
+                app_version: u16,
+            ) -> VersionizeResult<()> {
+                if self.len() > MAX_VEC_SIZE / std::mem::size_of::<$GenericParam>() {
+                    return Err(VersionizeError::VecLength(self.len()));
+                }
+
+                // Serialize in the same fashion as bincode:
+                // Write len.
+                bincode::serialize_into(&mut writer, &self.len())
+                    .map_err(|ref err| VersionizeError::Serialize(format!("{:?}", err)))?;
```

Comment:
Does `writer` needs to be marked as `mut`?
I see it is used for `bincode::serialize_into(&mut writer, ...`, but shouldn't just `writer` also work, because it is already a `&mut W`?